lib/aggregate/attribute/foreign_key.rb

```diff
@@ -1,25 +1,22 @@
 # frozen_string_literal: true
 
 class Aggregate::Attribute::ForeignKey < Aggregate::Attribute::Base
-
   def self.available_options
     super + [
       :class_name # The class for the foreign key
     ]
   end
 
   def from_value(value)
-    if value.is_a?(ActiveRecord::Base)
-      value
-    elsif value.nil?
-      nil
-    else
-      klass.find(value)
+    if value
+      Aggregate::ForeignKeyReference.new(klass, value)
     end
   end
 
   def from_store(value)
-    klass.find(value) if value
+    if value
```

Comment:
this prevents us from hitting the `ActiveRecord::RecordNotFound`..just wondering if we should bake this check into `Aggregate::ForeignKeyReference` so we don't always have to check it before newing up a reference